src/libpriv/rpmostree-core.cxx

```diff
@@ -1765,9 +1764,19 @@ rpmostree_context_prepare (RpmOstreeContext *self, GCancellable *cancellable, GE
   /* setup sack if not yet set up */
   if (dnf_context_get_sack (dnfctx) == NULL)
     {
+      auto flags = (DnfContextSetupSackFlags) (DNF_CONTEXT_SETUP_SACK_FLAG_LOAD_UPDATEINFO | DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELISTS);
+      for (auto &pkg_str : packages)
+        {
+          auto pkg = std::string (pkg_str).c_str ();
+          char *query = strchr((char*) pkg, '/');
+          if (query) {
+            flags = (DnfContextSetupSackFlags) (DNF_CONTEXT_SETUP_SACK_FLAG_LOAD_UPDATEINFO);
+          }
+        }
```

Comment:
I think the general logic of this makes sense, since it's pretty much the approach used in `dnf`: [PR](https://github.com/rpm-software-management/dnf5/pull/123/files#diff-3ac069fae0a65f95ddc5458710ca2abcd02322b7dc7e010ad42b10afc995a99cR243-R245). However, `rpmostree_context_prepare` will always find **all** layered packages and store them in the `packages` array, not just the ones currently being installed. For example, if you previously installed `/usr/bin/htop` and then installed `tmux`, the `packages` array will also contain `/usr/bin/htop`, disabling the `DNF_CONTEXT_SETUP_SACK_FLAG_SKIP_FILELIST` even though the current package being installed is not a filename path input. Not really sure if this is or isn't desired behaviour, but input would be appreciated here as well.